lib/puppet/defaults.rb

```diff
@@ -1464,7 +1464,21 @@ def self.valid_file_checksum_types
         with the running configuration.  In the case of puppet master,
         this file reflects the state discovered through interacting
         with clients."
-      },
+    },
+    :statettl => {
+      :default => "7d",
+      :type    => :ttl,
+      :desc    => "How long the Puppet agent should cache when a resource was last checked or synced.
+      environment.
+      #{AS_DURATION}
+      A value of `0`, or `unlimited` will disable cache pruning.
+
+      This setting affects the usage of Schedule resources, as the information
+      about when a resource was last checked (and therefore when it needs to be
+      checked again) is stored in the `statefile`. The `statettl` needs to be
+      large enough to ensure that a resource will not trigger multiple times
+      during a Schedule due to its cache entry ageing out of the cache."
```

Comment:
"ageing" should probably be "aging" or maybe instead of "aging out of" could just be replaced with "expiring from" if that would have the same meaning?